starlite/middleware/authentication.py

```diff
@@ -43,47 +42,34 @@ class AbstractAuthenticationMiddleware(ABC, MiddlewareProtocol):
     The status code to for websocket authentication errors.
     """
 
-    def __init__(self, app: "ASGIApp"):
+    def __init__(
+        self,
+        app: "ASGIApp",
+        exclude: Optional[Union[str, List[str]]] = None,
+    ):
         """This is an abstract AuthenticationMiddleware that allows users to
         create their own AuthenticationMiddleware by extending it and
         overriding the 'authenticate_request' method.
 
         Args:
             app: An ASGIApp, this value is the next ASGI handler to call in the middleware stack.
+            exclude: A pattern or list of patterns to skip in the authentication middleware.
         """
         super().__init__(app)
         self.app = app
+        self.exclude: Optional[Pattern[str]] = None
+        if exclude:
+            self.exclude = re.compile("|".join(exclude)) if isinstance(exclude, list) else re.compile(exclude)
 
     async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> None:
-        try:
+        if self.exclude and self.exclude.findall(scope["path"]):
+            await self.app(scope, receive, send)
+        else:
             if scope["type"] in self.scopes:
                 auth_result = await self.authenticate_request(HTTPConnection(scope))
                 scope["user"] = auth_result.user
                 scope["auth"] = auth_result.auth
             await self.app(scope, receive, send)
-        except (NotAuthorizedException, PermissionDeniedException) as e:
-            if scope["type"] == ScopeType.WEBSOCKET:  # pragma: no cover
-                await send({"type": "websocket.close", "code": self.websocket_error_status_code, "reason": repr(e)})
```

Comment:
this code path needs to be kept

Reply:
Isn't the `else` clause of the exception middleware pretty much this?

```python
            if scope["type"] == ScopeType.HTTP:
                ...
            else:
                status_code = (
                    exc.status_code if isinstance(exc, StarletteHTTPException) else HTTP_500_INTERNAL_SERVER_ERROR
                )
                # The 4000+ code range for websockets is customizable, hence we simply add it to the http status code
                status_code += 4000
                reason = repr(exc)
                await send({"type": "websocket.close", "code": status_code, "reason": reason})
```

Reply:
valid point. In this case - my points are mut. I will just remove the class vars, make a breaking change, and update the docs before finishing this.